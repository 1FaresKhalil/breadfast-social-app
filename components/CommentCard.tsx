import React from 'react';
import { View, Text } from 'react-native';
import { styled } from 'nativewind';
import UserAvatar from './UserAvatar';
import Animated, { FadeIn, FadeOut } from 'react-native-reanimated';

interface Comment {
  id: number;
  post_id: number;
  name: string;
  body: string;
}

interface CommentCardProps {
  comment: Comment;
}

const StyledView = styled(View);
const StyledText = styled(Text);

const CommentCard: React.FC<CommentCardProps> = ({ comment }) => {
  return (
    <Animated.View
      entering={FadeIn}
      exiting={FadeOut}
      style={{
        backgroundColor: 'white',
        padding: 16,
        marginBottom: 16,
        borderRadius: 12,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.1,
        shadowRadius: 8,
        transform: [{ scale: 1 }],
      }}
    >
      <StyledView className="flex-row items-center mb-2">
        <UserAvatar userId={comment.post_id} />
        <StyledView className="ml-3">
          <StyledText className="font-bold text-lg text-gray-900">
            {comment.name}
          </StyledText>
        </StyledView>
      </StyledView>
      <StyledText className="text-gray-700 leading-relaxed">
        {comment.body}
      </StyledText>
    </Animated.View>
  );
};

export default CommentCard;
