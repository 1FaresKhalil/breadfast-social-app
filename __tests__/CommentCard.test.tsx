import React from 'react';
import renderer from 'react-test-renderer';
import CommentCard from '../components/CommentCard';

describe('<CommentCard />', () => {
  it('renders correctly', () => {
    const comment = {
      id: 1,
      post_id: 1,
      name: 'Test User',
      body: 'This is a test comment.',
    };
    const tree = renderer.create(<CommentCard comment={comment} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
