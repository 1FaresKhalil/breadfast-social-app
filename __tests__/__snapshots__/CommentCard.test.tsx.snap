// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentCard /> renders correctly 1`] = `
<View
  collapsable={false}
  entering={[Function]}
  exiting={[Function]}
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 12,
      "marginBottom": 16,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 8,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    style={
      [
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "marginBottom": 8,
          },
        ],
      ]
    }
  >
    <View
      style={
        [
          [
            {
              "width": 40,
            },
            {
              "height": 40,
            },
            {
              "backgroundColor": "#d1d5db",
            },
            {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            {
              "overflow": "hidden",
            },
          ],
        ]
      }
    >
      <Image
        source={
          {
            "uri": "https://robohash.org/1.png",
          }
        }
        style={
          [
            [
              {
                "width": "100%",
              },
              {
                "height": "100%",
              },
            ],
          ]
        }
      />
    </View>
    <View
      style={
        [
          [
            {
              "marginLeft": 12,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            [
              {
                "fontWeight": "700",
              },
              {
                "fontSize": 18,
                "lineHeight": 28,
              },
              {
                "color": "#111827",
              },
            ],
          ]
        }
      >
        Test User
      </Text>
    </View>
  </View>
  <Text
    style={
      [
        [
          {
            "color": "#374151",
          },
        ],
      ]
    }
  >
    This is a test comment.
  </Text>
</View>
`;
